enhancements/update/cvo-log-level-api.md

```diff
@@ -0,0 +1,194 @@
+	CVOLogLevelDebug CVOLogLevel = "Debug"
+
+	// Trace is used when something went really badly and even more verbose logs are needed.  Logging every function call as part of a common operation, to tracing execution of a query.  In kube, this is probably glog=6.
+	CVOLogLevelTrace CVOLogLevel = "Trace"
+
+	// TraceAll is used when something is broken at the level of API content/decoding.  It will dump complete body content.  If you turn this on in a production cluster
+	// prepare from serious performance issues and massive amounts of logs.  In kube, this is probably glog=8.
+	CVOLogLevelTraceAll CVOLogLevel = "TraceAll"
+)
+```
+
+### Workflow Description
+
+Given a cluster administrator and a working cluster for which the administrator is responsible.
+
+**cluster administrator** is a human user responsible for managing the cluster.
+
+1. The cluster administrator notices an issue in the cluster and chooses to troubleshoot the issue.
+2. The cluster administrator after some troubleshooting notices that the logs of the Cluster Version Operator (CVO) might help.
+3. The cluster administrator notices that the logs are not detailed enough to troubleshoot the issue, and the administrator raises the log level from the default value `Normal` to the level `Trace` by simply modifying the `ClusterVersion` resource via the web console or by patching the resource by using the CLI.
+4. The cluster administrator fixes the issue.
+5. The cluster administrator notices that the CVO outputs too many logs for the administrator's liking.
+6. The cluster administrator lowers the log level of the CVO to the level `Normal`.
+7. The cluster administrator is now a happy cluster administrator.
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+See the section [Proposal](#proposal) for the proposed API changes.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The CVO needs to change the log level dynamically without redeploying itself. The CVO currently uses the package [`klog`](https://github.com/kubernetes/klog) for logging. This package does not provide a simple way of changing the log level dynamically. However, it is possible, and there already exists a function 
+[`SetLogLevel`](https://github.com/openshift/library-go/blame/8477ec72b72554c01afff4df9c6a90c0e492ea87/pkg/operator/loglevel/util.go#L63-L103) in the [`openshift/library-go`](https://github.com/openshift/library-go) repository to remedy this issue.
+
+#### Hypershift [optional]
+
+N/A
+
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
+
+N/A
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+
+Unit tests will be written to test if setting `operatorLogLevel` sets the respective logging in the CVO.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
```

Comment:
Can we say that the relevant components are the CVO and the ClusterVersion CRD, that these move close together on updates (and downgrades), and that the empty-string default means that skew on both the newer CVO and newer CRD sides functions as "use the default log level", which is a sane default?

Reply:
> ...and that the empty-string default means...

I got a little confused as the default value for the field `OperatorLogLevel` is `"Normal"`. Could you please elaborate a little more? I would appreciate it!

So that I understand it correctly. Do you mean by that the following scenarios:

* The CVO is at `N`, ClusterVersion CRD is at `N+1`
    * The CVO ignores any value in the CRD and uses a default log level because the CVO is not aware of the new field. 
* The CVO is at `N+1`, ClusterVersion CRD is at `N` 
    * The CVO gets an empty value for the `OperatorLogLevel` field because it doesn't exist. The CVO evaluates it as a default log level.

Reply:
I have updated the enhancement. Let me know if I have misunderstood anything or it makes sense, thank you.